Src/Test.UnitTests.Sarif.PatternMatcher/SpamTestRule.cs

```diff
@@ -29,11 +29,11 @@ internal class SpamTestRule : Skimmer<AnalyzeContext>, IOptionsProvider
         internal static Random s_random = new Random(s_seed);
 
         [ThreadStatic]
-        internal static TestRuleBehaviors s_testRuleBehaviors;
+        private static TestRuleBehaviors s_testRuleBehaviors;
 
         public SpamTestRule()
         {
-            if (s_testRuleBehaviors.HasFlag(TestRuleBehaviors.RaiseExceptionInvokingConstructor))
+            if (S_testRuleBehaviors.HasFlag(TestRuleBehaviors.RaiseExceptionInvokingConstructor))
```

Comment:
>S_testRuleBehaviors [](http://example.com/codeflow?start=16&length=19)

@schlaman-ms I didn't fully follow this change. The variable name is still s_testRuleBehaviors right? Why capitalize the S?

Reply:
Well the issue here is to eliminate the warning I took the automatic visual studio change.
It does look like all that changed is internal to private and then s_test... to S_test...
But s_test... still exists and is the backing field and the new property S_test... is added.
That's why it looks hard to follow and odd why the case change s_ --> S_

Reply:
S_ is not a convention for any standard that I'm aware of.

:) if we prefer s_ it needs to remain that way and you'll need to disable warnings.

Reply:
Addressed this in the prior comment about s_testRuleBehaviors.